### PR TITLE
fix metric learning

### PR DESCRIPTION
**Pull request by kebinC**
fix metric learning multi gpu training

### CODE REVIEW

File: fluid/metric_learning/losses/metrics.py [hunk 1]
```diff
@@ -17,3 +17,63 @@ def recall_topk(fea, lab, k = 1):
...
+    a2 = fluid.layers.square(feature)
+    a2 = fluid.layers.reduce_sum(a2, dim = 1)
+    #fluid.layers.Print(a2, summarize=10)
+    d = fluid.layers.elementwise_add(-2*ab, a2, axis = 0)
+    d = fluid.layers.elementwise_add(d, a2, axis = 1)
+    d = fluid.layers.reshape(d, shape = [-1, 1])
+    index = generate_index(batch_size, samples_each_class)
+    index_var = fluid.layers.create_global_var(shape=[batch_size*batch_size], value=0, dtype='int32', persistable=True)
+    index_var = fluid.layers.assign(index, index_var)
+    #index_var.stop_gradient = True
```
Comment (qingqing01) [inline fluid/metric_learning/losses/metrics.py]: Please remove the commented code.

File: fluid/metric_learning/losses/metrics.py [hunk 2]
```diff
@@ -17,3 +17,63 @@ def recall_topk(fea, lab, k = 1):
             res += 1.0
     res = res/len(fea)
     return res
+
+import subprocess
+import os
+def getgpunum():
```
Comment (qingqing01) [inline fluid/metric_learning/losses/metrics.py]: getgpunum -> get_gpu_num

File: fluid/metric_learning/losses/metrics.py (hunk 1 above)
Comment (kebinC) [reply fluid/metric_learning/losses/metrics.py]: done

File: fluid/metric_learning/losses/metrics.py (hunk 2 above)
Comment (kebinC) [reply fluid/metric_learning/losses/metrics.py]: done

**Review by qingqing01 — APPROVED** [review]

**Pull request merged**